shells/pipes-shell-2/pipe.js

```diff
@@ -82,6 +82,12 @@ const populateDispatcher = (dispatcher, api, composerFactory, storage, context)
     },
     spawn: async (msg, tid, bus) => {
       return api.marshalSpawnArc(msg, tid, bus);
+    },
+    pec: async (msg, tid, bus) => {
+      if (!bus.pecPorts[msg.id]) {
```

Comment:
There is only ever one bus, why not have `pecPorts` be owned by `pipe.js`.

Reply:
done